File: backend/app/models/skill_gap.rb
class SkillGap < ApplicationRecord
  has_many :learning_tasks, dependent: :destroy

  enum :status, { not_started: 0, in_progress: 1, done: 2 }

  validates :title, presence: true, length: { maximum: 255 }
  validates :category, presence: true
  validates :current_level, numericality: { greater_than_or_equal_to: 0 }, allow_nil: true
  validates :target_level, numericality: { greater_than_or_equal_to: 0 }, allow_nil: true
end

File: frontend/src/types/skillGap.ts
export type SkillGapStatus = 'not_started' | 'in_progress' | 'done'
export type LearningTaskStatus = 'todo' | 'doing' | 'done'

export type LearningTask = {
  id: number
  skill_gap_id: number
  title: string
  status: LearningTaskStatus
  due_date: string | null
  memo: string | null
  created_at: string
  updated_at: string
}

export type SkillGap = {
  id: number
  title: string
  category: string
  current_level: number | null
  target_level: number | null
  reason: string | null
  status: SkillGapStatus
  learning_tasks_count: number
  created_at: string
  updated_at: string
  learning_tasks?: LearningTask[]
}

export type SkillGapCreateInput = {
  title: string
  category: string
  current_level?: number | null
  target_level?: number | null
  reason?: string | null
  status?: SkillGapStatus
}
